### PR TITLE
clean up hypre dependency install

### PR DESCRIPTION
**Pull request by sslattery**
The hypre dependency was not being correctly installed so downstream packages would have had to included hypre in their `CMAKE_PREFIX_PATH` even though it is an implementation detail in Cajita. This PR adds proper install for this.

### CODE REVIEW

File: cmake/CabanaConfig.cmakein [hunk 1]
```diff
@@ -1,4 +1,5 @@
 include(CMakeFindDependencyMacro)
+set( CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/FindHYPRE.cmake;${CMAKE_MODULE_PATH}" )
```
Comment (junghans) [inline cmake/CabanaConfig.cmakein:2]:
```suggestion
list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}" )
```
It's a list point to directories to search for.

File: cmake/CabanaConfig.cmakein [hunk 2]
```diff
@@ -7,11 +8,13 @@ if(@MPI_FOUND@)
   find_dependency(MPI REQUIRED CXX)
   set(Cabana_ENABLE_MPI ON)
```
Comment (streeve) [inline cmake/CabanaConfig.cmakein:9]: Should we make this `CABANA_ENABLE_MPI` to match Kokkos?

File: cmake/CabanaConfig.cmakein [hunk 3]
```diff
@@ -20,4 +23,5 @@ endif()
 if(@ARBORX_FOUND@)
   find_dependency(ArborX REQUIRED)
```
Comment (streeve) [inline cmake/CabanaConfig.cmakein:24]: `CABANA_ENABLE_ARBORX`, matching MPI, would be helpful

File: cmake/CabanaConfig.cmakein [hunk 4]
```diff
@@ -1,23 +1,22 @@
...
 endif()
+set(Cabana_ENABLE_HEFFTE @Cabana_ENABLE_HEFFTE@)
 if(@HEFFTE_FOUND@)
   find_dependency(Heffte REQUIRED)
 endif()
+set(Cabana_ENABLE_ARBORX @Cabana_ENABLE_ARBORX@)
 if(@ARBORX_FOUND@)
   find_dependency(ArborX REQUIRED)
 endif()
+include("${CMAKE_CURRENT_LIST_DIR}/CajitaTargets.cmake")
```
Comment (junghans) [inline cmake/CabanaConfig.cmakein:21]: Just found that bug when build `exapm` with spack as well.

File: cmake/CabanaConfig.cmakein (hunk 3 above)
Comment (junghans) [reply cmake/CabanaConfig.cmakein:19]: All caps `CABANA`?

File: CMakeLists.txt [hunk 5]
```diff
@@ -73,6 +73,11 @@ Cabana_add_dependency( PACKAGE CLANG_FORMAT )
 
 # find hypre
 Cabana_add_dependency( PACKAGE HYPRE )
+if(HYPRE_FOUND)
+  install(FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindHYPRE.cmake
+    DESTINATION lib/cmake/Cabana )
```
Comment (junghans) [inline CMakeLists.txt:79]:
```suggestion
    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Cabana )
```

**Comment by sslattery** [comment]
Travis isn't reporting back but the downstream testing for external packages is breaking in some builds:
```
~/build/ECP-copa/Cabana/example/tutorial/01_hello_world/build ~/build/ECP-copa/Cabana/example/tutorial/01_hello_world
2815-- The C compiler identification is GNU 7.5.0
2816-- The CXX compiler identification is GNU 7.5.0
2817-- Check for working C compiler: /usr/lib/ccache/gcc
2818-- Check for working C compiler: /usr/lib/ccache/gcc -- works
2819-- Detecting C compiler ABI info
2820-- Detecting C compiler ABI info - done
2821-- Detecting C compile features
2822-- Detecting C compile features - done
2823-- Check for working CXX compiler: /usr/lib/ccache/g++
2824-- Check for working CXX compiler: /usr/lib/ccache/g++ -- works
2825-- Detecting CXX compiler ABI info
2826-- Detecting CXX compiler ABI info - done
2827-- Detecting CXX compile features
2828-- Detecting CXX compile features - done
2829-- Enabled Kokkos devices: SERIAL
2830-- Found MPI_CXX: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so (found version "3.1") 
2831-- Found MPI: TRUE (found version "3.1") found components:  CXX 
2832-- Configuring done
2833CMake Error at CMakeLists.txt:9 (add_executable):
2834  Target "HelloWorld" links to target "ArborX::ArborX" but the target was not
2835  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
2836  an ALIAS target is missing?
```
I'm surprised ArborX isn't getting picked up

**Comment by sslattery** [comment]
ah it was related to #245! @junghans you beat me to it

**Pull request merged**